Make torch.library.triton_op public

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141883
* __->__ #141880
* #141894

We've been using it privately for half a year and everything's been
good. This PR:
1. Makes torch.library.triton_op public
2. Renames capture_triton -> wrap_triton. We got feedback that no one
   knew what "capture triton" does.
3. Makes torch.library.wrap_triton public.

triton_op is used to construct a Python custom operator that may call 1+
triton kernels. Each of those triton kernels must be annotated with
wrap_triton.

Test Plan:
- existing tests

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

zou3519 committed:
Update

[ghstack-poisoned]

zou3519 committed:
Update

[ghstack-poisoned]

zou3519 committed:
Update

[ghstack-poisoned]

zou3519 committed:
Update on "Make torch.library.triton_op public"


We've been using it privately for half a year and everything's been
good. This PR:
1. Makes torch.library.triton_op public
2. Renames capture_triton -> wrap_triton. We got feedback that no one
   knew what "capture triton" does.
3. Makes torch.library.wrap_triton public.

triton_op is used to construct a Python custom operator that may call 1+
triton kernels. Each of those triton kernels must be annotated with
wrap_triton.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "Make torch.library.triton_op public"


We've been using it privately for half a year and everything's been
good. This PR:
1. Makes torch.library.triton_op public
2. Renames capture_triton -> wrap_triton. We got feedback that no one
   knew what "capture triton" does.
3. Makes torch.library.wrap_triton public.

triton_op is used to construct a Python custom operator that may call 1+
triton kernels. Each of those triton kernels must be annotated with
wrap_triton.

Test Plan:
- existing tests

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

albanD commented on torch/library.py:
```diff
@@ -32,6 +33,8 @@
     "register_vmap",
     "get_ctx",
     "custom_op",
+    "triton_op",
```
I can't find the doc being rendered for these on the website. Am I missing something ?

zou3519 replied:
Weird, let me dig into it

zou3519 replied:
RIP my browser cached the artifacts. If your browser does too then I'd recommend opening the docs page in incognito mode lol

albanD replied:
This is weird because putting triton_op in the search doesn't bring it up.
cc @svekars is that due to the new search? This sounds bad?

svekars replied:
> This is weird because putting triton_op in the search doesn't bring it up. cc @svekars is that due to the new search? This sounds bad?
The search goes to stable docs most likely (the indexed pages). I will add a PR to enable old search in docs.preview.

albanD replied:
Thanks!

zou3519 committed:
Update on "Make torch.library.triton_op public"


We've been using it privately for half a year and everything's been
good. This PR:
1. Makes torch.library.triton_op public
2. Renames capture_triton -> wrap_triton. We got feedback that no one
   knew what "capture triton" does.
3. Makes torch.library.wrap_triton public.

triton_op is used to construct a Python custom operator that may call 1+
triton kernels. Each of those triton kernels must be annotated with
wrap_triton.

Test Plan:
- existing tests

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

albanD reviewed (approved):
SGTM!
